File: src/main/java/org/taktik/connector/business/dmg/builders/RequestObjectBuilderFactory.java
package org.taktik.connector.business.dmg.builders;

import org.taktik.connector.technical.exception.TechnicalConnectorException;
import org.taktik.connector.technical.utils.ConfigurableFactoryHelper;

public final class RequestObjectBuilderFactory {
   public static final String PROP_REQUESTBUILDER_CLASS = "dmg.requestobjectbuilder.class";
   public static final String DEFAULT_REQUESTBUILDER_CLASS = "org.taktik.connector.business.dmg.builders.impl.RequestObjectBuilderImpl";
   private static ConfigurableFactoryHelper<RequestObjectBuilder> helperFactoryrequestBuilder = new ConfigurableFactoryHelper("dmg.requestobjectbuilder.class", "org.taktik.connector.business.dmg.builders.impl.RequestObjectBuilderImpl");

   public static RequestObjectBuilder getRequestObjectBuilder() throws TechnicalConnectorException {
      return (RequestObjectBuilder)helperFactoryrequestBuilder.getImplementation();
   }
}

File: src/main/kotlin/org/taktik/connector/business/wsconsent/builders/RequestObjectBuilderFactory.kt
package org.taktik.connector.business.wsconsent.builders

import org.taktik.connector.business.wsconsent.exception.WsConsentBusinessConnectorException
import org.taktik.connector.technical.exception.TechnicalConnectorException
import org.taktik.connector.technical.utils.ConfigurableFactoryHelper

object RequestObjectBuilderFactory {
	private val helperFactoryRequestObjectBuilder = ConfigurableFactoryHelper<RequestObjectBuilder>("org.taktik.connector.business.wsconsent.builders.RequestObjectBuilder.class", "org.taktik.connector.business.wsconsent.builders.impl.RequestObjectBuilderImpl")
	private val helperFactoryConsentBuilder = ConfigurableFactoryHelper<ConsentBuilder>("wsconsent.ConsentBuilder.class", "org.taktik.connector.business.wsconsent.builders.impl.ConsentBuilderImpl")

	val requestObjectBuilder: RequestObjectBuilder
		@Throws(TechnicalConnectorException::class, WsConsentBusinessConnectorException::class, InstantiationException::class)
		get() = helperFactoryRequestObjectBuilder.getImplementation()

	val consentBuilder: ConsentBuilder
		@Throws(TechnicalConnectorException::class, WsConsentBusinessConnectorException::class)
		get() = helperFactoryConsentBuilder.getImplementation()

}
